src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -382,6 +429,21 @@ private static Block RebuildCodeGenerators(Block block)
             return new KeyValuePair<string, SyntaxTreeNode>(name, builder.Build());
         }
 
+        private static bool IsNullOrWhitespaceAttributeValue(SyntaxTreeNode attributeValue)
+        {
+            if (attributeValue.IsBlock)
+            {
+                attributeValue = ((Block)attributeValue).FindFirstDescendentSpan();
+            }
+
+            return attributeValue == null || string.IsNullOrWhiteSpace(((Span)attributeValue).Content);
+        }
```

Comment:
thanks for spreading this out.

I suspect this will cause errors for legal attribute values such as `" true "` or `" @true "`.  should handle the `Block` case using
``` c#
  foreach (var span in ((Block)attributeValue).Flatten())
  {
    if (!string.IsNullOrWhiteSpace(span.Content))
    {
      return false;
    }
  }
  return true;
```

if it's not necessary to check every `Span` in the `Block`, need a code comment explaining why.